mhs-reference-implementation/mhs/routing/sds.py

```diff
@@ -27,91 +23,95 @@
-                            f"ods: {ods_code} - interaction: {interaction_id}")
-        # As per the spec exactly one result should be returned
+            logger.warning(f"More than one accredited system details returned on inputs: "
+                           f"ods: {ods_code} - interaction: {interaction_id}")
 
+        # As per the spec exactly one result should be returned
         response = accredited_system_lookup[0]
         party_key = response['attributes'][MHS_PARTY_KEY]
 
         details = await self._mhs_details_lookup(party_key, interaction_id)
 
+        if not details:
+            logger.error(f'No mhs details returned for party key: {party_key} and interaction id : {interaction_id}')
+            raise routing_exception.RoutingException(f'No mhs details returned for party key: '
+                                                     f'{party_key} and interaction id : {interaction_id}')
         if len(details) > 1:
-            logging.warning(f"More than one mhs details  returned on inputs: "
-                            f"ods: {ods_code} - interaction: {interaction_id}")
+            logger.warning(f"More than one mhs details returned on inputs: "
+                           f"ods: {ods_code} - interaction: {interaction_id}")
         return details[0]['attributes']
 
-    async def _accredited_system_lookup(self, ods_code: str, interaction_id: str):
+    async def _accredited_system_lookup(self, ods_code: str, interaction_id: str) -> List:
         """
         Used to find an accredited system, the result contains the nhsMhsPartyKey.
         This can then be used to find an MHS endpoint
         :return: endpoint details - filtered to only contain nhsMHSPartyKey
         """
-        if not self.connection:
-            raise IOError('No connection established')
 
         search_filter = f"(&(nhsIDCode={ods_code}) (objectClass={AS_OBJECT_CLASS}) (nhsAsSvcIA={interaction_id}))"
 
         message_id = self.connection.search(search_base=NHS_SERVICES_BASE,
                                             search_filter=search_filter,
-                                            attributes="nhsMHSPartyKey")
+                                            attributes=MHS_PARTY_KEY)
+        logger.info(f'Message id - {message_id} - for query: ods code - {ods_code} '
+                    f': interaction id - {interaction_id}')
+
+        response = await self._get_query_result(message_id)
+        logger.info(f'Found accredited supplier details for message_id: {message_id}')
 
-        query_task = self._get_query_result(message_id)
-        response = await query_task
         return response
 
-    async def _mhs_details_lookup(self, party_key: str, interaction_id: str):
+    async def _mhs_details_lookup(self, party_key: str, interaction_id: str) -> List:
         """
         Given a party key and a service, this will return an object containing the attributes of that party key,
```

Comment:
I'm assuming interaction id and service mean the same thing (not familiar with all this terminology yet).

Reply:
This is correct, I will change this as Brian has specified previously that we should try to stick to interaction id, must have missed this one!